single_kernel_mongo/events/tls.py

```diff
@@ -64,36 +59,24 @@ def __init__(self, dependent: OperatorProtocol):
         self.framework.observe(
             self.certs_client.on.certificate_expiring, self._on_certificate_expiring
         )
+        self.framework.observe(self.charm.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.charm.on.secret_changed, self._on_secret_changed)
 
-    def _on_set_tls_private_key(self, event: ActionEvent) -> None:
-        """Set the TLS private key which will be used for requesting the certificates."""
-        logger.debug("Request to set TLS private key received.")
-        if (
-            self.manager.state.is_role(MongoDBRoles.MONGOS)
-            and self.manager.state.config_server_name is None
-        ):
-            logger.info(
-                "mongos is not running (not integrated to config-server) deferring renewal of certificates."
-            )
-            event.fail("Mongos cannot set TLS keys until integrated to config-server.")
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Validate private keys and refresh certs if needed."""
+        if self.manager.state.tls_relation is None:
```

Comment:
I think we should always run the update of the key, otherwise we lose that key if we relate after we set the private key.